### DISCUSSION TITLE
Problems changing branches

### DISCUSSION THREAD

**Discussion by RandomLuck-7** [community/community#59772] [Repositories] [answered]
### Select Topic Area

Question

### Body

I'm having trouble switching branches. Does not seem to be possible for me. I've been told that I need to change the "Default  Branch" But my menus do not match the menus on the how-to-do pages. No pull down menu of branches. No way to change another repository to the default branch. 

My menu options seem to be different from others. _Is there some setting I have wrong that is causing this problem?_ 

Thanks.

![Switch-defaul-branch-button](https://github.com/community/community/assets/52889245/3d4558b3-fb81-40d0-9a45-f9f09246e3ad)
![View-all-branches](https://github.com/community/community/assets/52889245/33b4a218-ab8d-45dc-8c7d-d9e2fe4a99fb)
![my-available-options-w-xdripswift-repository](https://github.com/community/community/assets/52889245/1d8e3d7d-ec8e-48ee-a804-7cd7d41f1c32)

**Comment by 3urobeat** [comment]
Hey!  
Switching branches and changing the default branch are two different things.  
The "Switch branch" dropdown in your second screenshot allows you to select a different branch to view it. Since you only have one branch in your repository you are only seeing the one already selected.

Switching the default branch, as shown in your 3rd screenshot, changes the branch anyone will see at first when visiting your repository. Since you only have one branch in your repo you only see the "Rename branch" button. This allows you to rename that one branch, which is already your default branch.  
If you want to change the default branch you first need to create another branch (to be able to change it something else). Another button will then pop up right beside your "Rename branch" button, called "Switch to another branch". Clicking on that button will open the dialog with a dropdown you probably are searching for.

I hope this helps you

**Reply by 3urobeat** [reply, marked as answer]
One more thing, seeing that this is a fork - have you maybe selected the "Copy the main branch only" checkbox when forking? This means that only the default branch was forked.  
If you want all branches of the parent repository to be able to "change the default branch" to another existing branch of the parent repo, you must not click that checkbox when forking.

See here for a screenshot of what I mean: https://github.blog/changelog/2022-07-27-you-can-now-fork-a-repo-and-copy-only-the-default-branch/

**Reply by RandomLuck-7** [reply]
**_That_** could be the problem! (I have 4 different repositories that are working), but I was not aware of the checkbox when forking.

**Comment by GhostOf0days** [comment]
To create a new branch using terminal (preferred), make sure git is downloaded and do ```git checkout -b [branch-name]```. This creates a new branch with the name you specified and switches you to that new branch. You could also go to where you see "Find or create a branch" in your first image, type a branch name, and press enter. This method is not preferred. Learning to use git in the terminal or even a combination of using git in the terminal in addition to something like Github Desktop is better than using the Github website for these things. Just as advice, for every new issue you want to fix, create a new branch from your main branch. Make sure your main branch is up-to-date then do ```git checkout main``` followed by ```git checkout -b [branch-name]```. This also will pull all your code from the main branch into your new branch, so you can work on relevant issues. After you are done with fixing an issue, you can create a PR (pull request; using the Github website is not bad here) and merge the relevant changes into the main branch (make sure if multiple people are on the project, someone reviews it rather than just committing a fix to an issue. Also, adding an issue in the setting for the PR that says "Development" will automatically close that issue upon merging your changes.

**Reply by GhostOf0days** [reply]
I assume this is what you intended to do rather than renaming the main branch as you don't have any other branches. A good resource to learn using branching is this: https://learngitbranching.js.org/

**Reply by GhostOf0days** [reply]
There are also tools like GitLens that may help you better understand branching.

**Comment by Shivam-netweb** [comment]
I accidently creates a pull request into wrong branch, but when it try to change the branch by editing it, it doesn't open the branch dropdown there.

for instance I've uploaded a screenshot for it

![image](https://github.com/community/community/assets/102536014/5584001d-f39d-40f6-b324-313a75b07711)
